python_modules/dagster/dagster/_core/execution/execution_result.py

```diff
@@ -39,7 +39,9 @@ def run_id(self) -> str:
     @property
     def success(self) -> bool:
         """bool: Whether execution was successful."""
-        return self.dagster_run.is_success
+        return self.dagster_run.is_success and not self.filter_events(
+            lambda evt: evt.is_step_failure or evt.is_run_failure
+        )
```

Comment:
this seems extremely dubious - how is the run successful and the events failures?